xbmc/platform/android/network/NetworkAndroid.cpp

```diff
@@ -279,7 +279,7 @@ CNetworkInterface* CNetworkAndroid::GetFirstConnectedInterface()
 {
   std::unique_lock<CCriticalSection> lock(m_refreshMutex);
 
-  if (CJNIBase::GetSDKVersion() >= 24)
+  if (CJNIBase::GetSDKVersion() >= 24 && m_defaultInterface)
```

Comment:
This condition can be simplified to:

` if (m_defaultInterface)` 

because in SDK < 24 guarantees that `m_defaultInterface`  never is initialized then is redundant check SDK here.

Reply:
Right, more simple